### PR TITLE
Add shared-as name to instructor question preview

### PR DESCRIPTION
**Pull request by SethPoulsen**
This will make it easier for people to copy-paste the qid (with sharing name) of the question they are sharing, to give to other instructors, instead of having to construct it manually. It will also make it easier for instructors to link to the publicly shared version of their publicly shared questions.

### CODE REVIEW

**Review by nwalters512 — DISMISSED** [review]
`instructorInfoPanel` is used on more than just the `instructorQuestionPreview` page. For instance, this breaks the instance question page when an instructor is emulating a student:

Screenshot 2024-04-23 at 14 13 28

**Comment by SethPoulsen** [comment]
oof. 

Should I change it just to check if the variable `question_is_shared` exists, before checking if it is true? or something more complicated?

**Comment by nwalters512** [comment]
I think checking `typeof question_is_shared !== 'undefined'` would be sufficient here.

**Comment by SethPoulsen** [comment]
I suppose an alternative would be to move the code that attaches `question_is_shared` to `res.locals` to happen instead in the shared library code that all those pages use. Would that be better?

File: apps/prairielearn/src/pages/partials/instructorInfoPanel.ejs [hunk 1]
```diff
@@ -50,10 +50,23 @@
...
-      <div class="pr-1">Title:</div>
-      <div><%= question.title %></div>
-    </div>
+
+  <% if (typeof question_is_shared !== 'undefined' && question_is_shared) { %>
+  <div class="d-flex flex-wrap">
+    <div class="pr-1">Shared As:</div>
+    <% if (question.shared_publicly) { %>
+      <div>
+        <a href="<%= plainUrlPrefix %>/public/course/<%= course.id %>/question/<%= question.id %>/preview?variant_seed=<%= variant.variant_seed %>">@<%= course.sharing_name %>/<%= question.qid %></a>
```
Comment (SethPoulsen) [inline apps/prairielearn/src/pages/partials/instructorInfoPanel.ejs:59]: do we care to preserve the variant see here or no? 

It makes sense when linking back to the same page you are already on, but when linking from the private instructor preview, to the public preview, do we want it?

Comment (mwest1066) [reply apps/prairielearn/src/pages/partials/instructorInfoPanel.ejs:59]: I'm inclined to not preserve the variant seed.

Comment (SethPoulsen) [reply apps/prairielearn/src/pages/partials/instructorInfoPanel.ejs]: done

**Comment by jonatanschroeder** [comment]
This is currently not showing in the public question page itself...

**Comment by SethPoulsen** [comment]
It is! the Public question preview page has its own separate info panel.

**Comment by jonatanschroeder** [comment]
> It is! the Public question preview page has its own separate info panel.

The link yes, but the link has the QID as the text instead of the share name.

**Comment by jonatanschroeder** [comment]
image

**Comment by jonatanschroeder** [comment]
Found the culprit (from #9604):

https://github.com/PrairieLearn/PrairieLearn/blob/c364ca4093f6048e8ea856b4904f675292a541af/apps/prairielearn/src/pages/publicQuestionPreview/publicQuestionPreview.html.ts#L65-L69

**Comment by SethPoulsen** [comment]
Ok, so its only on the example course where it doesn't show it--and that is by design. 

So this PR is good to go then? or are there any other concerns?

**Review by jonatanschroeder — APPROVED** [review]

**Pull request merged**